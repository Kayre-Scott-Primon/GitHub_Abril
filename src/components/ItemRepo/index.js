import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { useDispatch } from 'react-redux';
import { setRespository } from '../../storage/redux/reducer/Repository';
import Styles from './styles';

export default function ItemRepo({data}) {

    const dispatch = useDispatch()
    const navigation = useNavigation()

    async function onPressRepo(){
        const repo = {
            title: data.name,
            url: data.html_url,
            owner: data.owner.login,
            stars: data.stargazers_count
        }
        await dispatch(setRespository(repo))
        navigation.navigate('Repository')
    }

    return (
        <Styles.Container
            onPress={onPressRepo}
        >
            <Styles.ContainerAvatarTitle>
                <Styles.Avatar     
                    testID='avatar'
                    source={{
                        uri: data.owner.avatar_url,
                    }}/>
                <Styles.TitleContainer>
                    <Styles.Title testID='name'>{data.name}</Styles.Title>
                    <Styles.Subtitle testID='owner'>{data.owner.login}</Styles.Subtitle>
                </Styles.TitleContainer>
            </Styles.ContainerAvatarTitle>
            <Styles.NumOfStar testID='stars'>{data.stargazers_count}</Styles.NumOfStar>
        </Styles.Container>
    )
}
