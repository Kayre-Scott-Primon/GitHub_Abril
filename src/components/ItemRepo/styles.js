import styled from 'styled-components/native';
import { colors } from '../../styles/colors';

export  default Styles = {
    Container: styled.TouchableOpacity`
        width: 100%;
        flex-direction: row;
        justify-content: space-between;
    `,

    ContainerAvatarTitle: styled.View`
        flex-direction: row;
        flex: 8;
    `,

    Avatar: styled.Image`
        width: 52px;
        height: 52px;
        background: ${colors.avatar};
        border-radius: 50px;
        margin-right: 12px;
    `,

    TitleContainer: styled.View`
        flex: 1;
        margin-right: 5px;
    `,

    Title: styled.Text.attrs({
        numberOfLines: 1
    })`
        font-family: 'SF Pro Text';
        font-style: normal;
        font-weight: 600;
        font-size: 16px;
        line-height: 24px;
        letter-spacing: -0.38px;
        color: ${colors.text};
        margin-bottom: 1px;
    `,

    Subtitle: styled.Text.attrs({
        numberOfLines: 1
    })`
        font-family: 'SF Pro Text';
        font-style: normal;
        font-weight: 400;
        font-size: 14px;
        line-height: 24px;
        color: ${colors.subtitle};
    `,

    NumOfStar: styled.Text`
        font-family: 'SF Pro Text';
        font-style: normal;
        font-weight: 400;
        font-size: 14px;
        line-height: 18px;
        letter-spacing: -0.078px;
        color: ${colors.subtitle};
        flex: 1;
    `

}