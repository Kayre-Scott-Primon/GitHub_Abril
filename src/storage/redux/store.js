import { configureStore } from "@reduxjs/toolkit";
import repository from "./reducer/repositorySlice";
import createSagaMiddleware from 'redux-saga'
import saga from './saga/saga'

let sagaMiddleware = createSagaMiddleware()
const middleware = [sagaMiddleware]

export default configureStore({
    reducer: {
        repository: repository
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(middleware),
})

sagaMiddleware.run(saga)
