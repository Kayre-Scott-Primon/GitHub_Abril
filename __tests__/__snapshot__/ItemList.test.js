import React from "react";
import renderer from 'react-test-renderer';
import ItemRepo from '../../src/components/ItemRepo'
import { Provider } from "react-redux";
import store from "../../src/storage/redux";

const mockedDispatch = jest.fn();

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
        ...actualNav,
        useNavigation: () => ({
        navigate: jest.fn(),
        dispatch: mockedDispatch,
        }),
    };
});

describe('ItemRepoSnapshot', () => {

  // genarate snapshot file
    /*it('renders correctly when there are no items', () => {
      const tree = renderer.create(            
        <Provider store={store}>
          <ItemRepo
              data={{
                  name: 'name',
                  url: 'url',
                  owner: {
                      login: 'user'
                  },
                  stargazers_count: 0
              }}
          />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });*/

    it('Verificando igualdade na estilização', () => {
      const tree = renderer.create(
        <Provider store={store}>
          <ItemRepo
              data={{
                  name: 'name',
                  url: 'url',
                  owner: {
                      login: 'user'
                  },
                  stargazers_count: 0
              }}
          />
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

})