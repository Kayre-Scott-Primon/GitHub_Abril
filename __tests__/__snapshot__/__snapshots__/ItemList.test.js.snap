// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemRepo renders correctly when there are no items 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 8,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Image
      source={
        {
          "uri": undefined,
        }
      }
      style={
        [
          {
            "backgroundColor": "#D9D9D9",
            "borderRadius": 50,
            "height": 52,
            "marginRight": 12,
            "width": 52,
          },
        ]
      }
      testID="avatar"
    />
    <View
      style={
        [
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginRight": 5,
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#000",
              "fontFamily": "SF Pro Text",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "600",
              "letterSpacing": -0.38,
              "lineHeight": 24,
              "marginBottom": 1,
            },
          ]
        }
        testID="name"
      >
        name
      </Text>
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#48484A",
              "fontFamily": "SF Pro Text",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 24,
            },
          ]
        }
        testID="owner"
      >
        user
      </Text>
    </View>
  </View>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "color": "#48484A",
          "flexBasis": 0,
          "flexGrow": 2,
          "flexShrink": 1,
          "fontFamily": "SF Pro Text",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": -0.078,
          "lineHeight": 18,
          "marginRight": 4,
          "textAlign": "right",
        },
      ]
    }
    testID="stars"
  >
    0
  </Text>
</View>
`;

exports[`ItemRepo renders correctly when there is a single item 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 8,
          "flexShrink": 1,
        },
      ]
    }
  >
    <Image
      source={
        {
          "uri": undefined,
        }
      }
      style={
        [
          {
            "backgroundColor": "#D9D9D9",
            "borderRadius": 50,
            "height": 52,
            "marginRight": 12,
            "width": 52,
          },
        ]
      }
      testID="avatar"
    />
    <View
      style={
        [
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginRight": 5,
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#000",
              "fontFamily": "SF Pro Text",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "600",
              "letterSpacing": -0.38,
              "lineHeight": 24,
              "marginBottom": 1,
            },
          ]
        }
        testID="name"
      >
        name
      </Text>
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#48484A",
              "fontFamily": "SF Pro Text",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 24,
            },
          ]
        }
        testID="owner"
      >
        user
      </Text>
    </View>
  </View>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "color": "#48484A",
          "flexBasis": 0,
          "flexGrow": 2,
          "flexShrink": 1,
          "fontFamily": "SF Pro Text",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": -0.078,
          "lineHeight": 18,
          "marginRight": 4,
          "textAlign": "right",
        },
      ]
    }
    testID="stars"
  >
    0
  </Text>
</View>
`;
